### DISCUSSION TITLE
PhotoDemon freezes up on large psd files

### DISCUSSION THREAD

**Discussion by phxjw** [tannerhelland/PhotoDemon#446] [General]
I much prefer the simplicity of PhotoDemon to photoshop but have noticed that it freezes up on larger sized psd files.  Approx 30mb or higher.  And it if doesn't freeze up, reaction time within the program can be slow.  I have neither of these issues with adobe photoshop so I'm inclined to believe it's not due to my own computer specs.  Any suggestions?

**Comment by tannerhelland** [comment]
Hi @phxjw .  Can you attach the PSD file that gives you trouble?  It's really hard to comment on performance without the file to examine.  As a 32-bit portable application, PhotoDemon typically works best on files with dimensions <= ~16 megapixels , and of course multiple layers impact performance accordingly.  (For reference, a 4K image is about half that size - 4096x2160 or 8.8 megapixels.)

I'll also mention up front that I don't want to sound dismissive... but I'd absolutely expect Photoshop to handle PSDs better than 3rd-party software like mine.  Photoshop is developed by a Fortune 500 company with ~25,000 employees!  I'm a solo developer who has had to manually reverse-engineer PSDs just to get them to work at all 😉 .  Indie developers like me do our best, but there will always be limits to how closely we can approximate a behemoth like Photoshop.

**Reply by phxjw** [reply]
No worries on sounding dismissive.  I should have prefaced my post by telling you how much I, and presumably others, appreciate the hard work you put into PhotoDemon.  Especially considering that you have provided it free of charge.  That being said, I do plan on making a donation at some point to help you continue with the program.  As said before, I prefer PhotoDemon and it's easier to use interface over other free and paid options out there that seem very convoluted.  Having a larger file not work within it is hardly something for me to get too worked up about.  Thank you again for your hard work and the quick reply.

**Reply by tannerhelland** [reply]
Thank you for the very kind reply, @phxjw  🙏 

If you continue to encounter slowdowns with PSD files, please do attach the problematic one(s) here (you should be able to click+drag a PSD file into a GitHub comment window to attach it, or if that doesn't work stick it in a .zip file and try attaching that) and I'll investigate further.  

In the meantime, I have started work on a new round of performance improvements for the next release that may alleviate some of the issues you've been facing.  Hopefully we can make some meaningful improvements one way or another!
